keras_nlp/metrics/rouge_l.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ROUGE-L metric implementation based on `keras.metrics.Metric`."""
+
+import tensorflow as tf
+import tensorflow_text as tf_text
+from tensorflow import keras
+
+
+class RougeL(keras.metrics.Metric):
+    """ROUGE-L metric.
+
+    This class implements the ROUGE-L metric.
+
+    Args:
+        alpha: float. `alpha` is used as the weight for the
+            harmonic mean of precision and recall  (for calculating F1-score). A
+            value of 0 means recall is more important and a value of 1 means
+            precision is more important (same behaviour as
+            https://www.tensorflow.org/text/api_docs/python/text/metrics/rouge_l).
+        metric_type: string. One of "precision", "recall", "f1_score". Defaults
+            to "f1_score".
+        mask_token_ids: list of integers. IDs of the tokens to be masked.
+        dtype: string or tf.dtypes.Dtype. Precision of metric computation. If
+            not specified, it defaults to tf.float32.
+        name: string. Name of the metric instance.
+        **kwargs: Other keyword arguments.
+
+    Examples:
+
+    1. Calculate RougeL (F1 Score) by calling `update_state()` and `result()`.
+    1.1. `mask_token_ids` not provided.
+    >>> tf.random.set_seed(42)
+    >>> rouge_l = keras_nlp.metrics.RougeL(name="rouge_l")
+    >>> references = tf.random.uniform(
```

Comment:
Prefer use a well-defined example rather than random data so that users can manually calculate the F1 score.